jcl/src/java.base/share/classes/java/lang/StringBuffer.java

```diff
@@ -1337,7 +1346,7 @@ public synchronized StringBuffer replace(int start, int end, String string) {
 		} else {
 			// Check if the StringBuffer is compressed
 			if (count >= 0) {
-				decompress(value.length);				
+				decompress(value.length, currentLength);
```

Comment:
There are three copies of the code on lines 1303-1305: suggest hoisting them before the `if` on line 1299.